to.etc.domui/src/main/java/to/etc/domui/server/OopsFrameRenderer.java

```diff
@@ -125,41 +133,46 @@ static private void dumpSingle(@NonNull StringBuilder sb, @NonNull Throwable x,
 		}
 	}
 
-
 	private static int findName(@NonNull List<StackTraceElement> list, String name) {
-		for(int i = list.size(); --i >= 0;) {
+		for(int i = list.size(); --i >= 0; ) {
 			String cn = list.get(i).getClassName();
 			if(name.equals(cn))
 				return i;
 		}
 		return -1;
 	}
 
-	private static List<StackTraceElement> stripFrames(@NonNull List<StackTraceElement> list, int from) {
+	private List<StackTraceElement> stripFrames(@NonNull List<StackTraceElement> list, int from) {
 		return list.subList(0, from - 1);
 	}
 
-	private static void appendTraceLink(@NonNull StringBuilder sb, @NonNull StackTraceElement ste) {
-		sb.append("        <a class='exc-stk-l' href=\"#\" onclick=\"linkClicked('");
+	private void appendTraceLink(@NonNull StringBuilder sb, @NonNull StackTraceElement ste) {
+		if(m_stacktraceTag == "a") {
```

Comment:
That is an incorrect "if". The string "a" is a reference type, and m_stacktraceTag can easily point to another instance of the same value "a". In that case the if would fail. You need to use equals.
Using a tag name in this way also makes the code fragile because the decision what to render is for 99% here- but you have that tagname set outside. If you see how many ifs are inside here it is way better to just have a boolean m_testMode field and do the ifs on that. You can then make a local variable "tagName" in here if you really want (although for me doing without it would still be cleaner).
The same goes for the template name too, actually- way better to just select the correct template depending on m_testMode inside getExceptionTemplate().